### PR TITLE
Fix OvPhysX cloning to preserve nested asset poses

### PR DESCRIPTION
**Pull request by marcodiiga**
# Description

OVPhysX clone transforms position each exact target subtree root at its final
world pose. Some shipped wording described parent poses; the corresponding
OVPhysX API name and documentation are being corrected separately. Isaac Lab
passes this argument positionally, so the adapter remains compatible.

The Isaac Lab adapter previously forwarded each environment origin for every
clone row, which dropped nested asset offsets and orientations. This change
composes each source row's environment-relative pose with the target
environment pose before calling `PhysX.clone()`. Environment-root cloning
remains unchanged.

Optional pose tensors are copied to CPU lists once before the setup-time target
loop, avoiding per-component device synchronization.

No linked public issue.

## Type of change

- Bug fix (non-breaking change which fixes an issue)

## Screenshots

Not applicable.

## Checklist

- [x] I have read and understood the [contribution guidelines](https://isaac-sim.github.io/IsaacLab/main/source/refs/contributing.html)
- [x] I have run the [`pre-commit` checks](https://pre-commit.com/) with `./isaaclab.sh --format`
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added a changelog fragment under `source/<pkg>/changelog.d/` for every touched package (do **not** edit `CHANGELOG.rst` or bump `extension.toml` - CI handles that)
- [x] I have added my name to the `CONTRIBUTORS.md` or my name already exists there

### CODE REVIEW

File: source/isaaclab_ovphysx/isaaclab_ovphysx/cloner/replicate.py [hunk 1]
```diff
@@ -82,41 +110,60 @@ def queue_mapping(
...
                     pos = positions[env_id]
-                    parent_positions.append((float(pos[0]), float(pos[1]), float(pos[2])))
-                else:
-                    parent_positions.append((0.0, 0.0, 0.0))
+                    target_env_world.SetTranslateOnly(Gf.Vec3d(float(pos[0]), float(pos[1]), float(pos[2])))
+                if quaternions is not None and env_id < len(quaternions):
+                    quat = quaternions[env_id]
+                    target_env_world.SetRotateOnly(
+                        Gf.Quatd(
+                            float(quat[3]),
```
[the remark this answers is not in the document]
Comment (marcodiiga) [reply source/isaaclab_ovphysx/isaaclab_ovphysx/cloner/replicate.py:187]: Confirmed xyzw. The sibling USD replicator consumes this argument as `Gf.Quatd(q[3], Gf.Vec3d(q[0], q[1], q[2]))`, and `grid_transforms()` documents and returns identity as `(0, 0, 0, 1)`. The OVPhysX clone pose is likewise `(px, py, pz, qx, qy, qz, qw)`. The updated regression uses four distinct quaternion components so a permutation now fails.

File: source/isaaclab_ovphysx/test/test_cloner.py [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    )
+
+    assert len(context._queue) == 1
+    source, targets, target_transforms = context._queue[0]
+    assert source == "/World/envs/env_0/Robot"
+    assert targets == ["/World/envs/env_1/Robot"]
+    assert len(target_transforms) == 1
+    assert target_transforms[0][:3] == pytest.approx((9.0, 20.0, 32.0))
+    orientation = torch.tensor(target_transforms[0][3:])
+    expected_orientation = torch.full((4,), 0.5)
```
[the remark this answers is not in the document]
Comment (marcodiiga) [reply source/isaaclab_ovphysx/test/test_cloner.py]: Addressed: the test now composes distinct X and Z rotations, sign-aligns equivalent `q` / `-q` values, and compares every xyzw component.

File: source/isaaclab_ovphysx/isaaclab_ovphysx/cloner/replicate.py [hunk 3]
```diff
@@ -82,41 +110,60 @@ def queue_mapping(
...
+            if not active_env_ids:
+                continue
 
             self_env_id: int | None = None
             pre, suf = split_clone_template(destinations[i])
             candidate = src.removeprefix(pre).removesuffix(suf)
             if candidate.isdigit():
                 self_env_id = int(candidate)
 
+            source_world = xform_cache.GetLocalToWorldTransform(self.stage.GetPrimAtPath(src)).RemoveScaleShear()
```
[the remark this answers is not in the document]
Comment (marcodiiga) [reply source/isaaclab_ovphysx/isaaclab_ovphysx/cloner/replicate.py]: In the supported ClonePlan path, every active row source is already authored on the stage, while inactive rows return before stage access. When the template yields a numeric environment id, the anchor is an ancestor of that valid source and cannot fail independently. If that invariant changes, validation should be added at plan/path construction; I am keeping this targeted fix scoped.

File: source/isaaclab_ovphysx/isaaclab_ovphysx/cloner/replicate.py [hunk 4]
```diff
@@ -82,41 +110,60 @@ def queue_mapping(
...
                     continue
                 targets.append(destinations[i].format(env_id))
+
+                target_env_world = Gf.Matrix4d(1.0)
                 if positions is not None and env_id < len(positions):
                     pos = positions[env_id]
-                    parent_positions.append((float(pos[0]), float(pos[1]), float(pos[2])))
-                else:
-                    parent_positions.append((0.0, 0.0, 0.0))
+                    target_env_world.SetTranslateOnly(Gf.Vec3d(float(pos[0]), float(pos[1]), float(pos[2])))
```
[the remark this answers is not in the document]
Comment (marcodiiga) [reply source/isaaclab_ovphysx/isaaclab_ovphysx/cloner/replicate.py:182]: Addressed: each optional tensor is detached and copied to a CPU list once before iteration, removing per-component device synchronization. The remaining target loop constructs paths and Gf matrices once during scene setup.

File: source/isaaclab_ovphysx/isaaclab_ovphysx/cloner/replicate.py [hunk 5]
```diff
@@ -82,41 +110,60 @@ def queue_mapping(
...
+                if quaternions is not None and env_id < len(quaternions):
```
[the remark this answers is not in the document]
Comment (marcodiiga) [reply source/isaaclab_ovphysx/isaaclab_ovphysx/cloner/replicate.py]: Inputs are documented as per-environment tensors aligned with `env_ids`; the supported ClonePlan path supplies full-size positions, while `None` intentionally means identity. The short-position fallback predates this PR. Adding shape-validation exceptions for direct callers would be a broader contract change, so I am keeping this fix scoped.

File: source/isaaclab_ovphysx/isaaclab_ovphysx/physics/ovphysx_manager.py [hunk 6]
```diff
@@ -253,13 +252,18 @@ def register_clone(
...
-                Xform prim (e.g. /World/envs/env_N).  When provided the clone
-                plugin sets those transforms in Fabric so all environments start
-                at their correct grid locations, preventing solver divergence
-                during the warmup step.
+            parent_positions: Final world positions (x, y, z) [m] for whole-environment
+                target roots. Each position uses an identity rotation.
         """
-        cls._pending_clones.append((source, targets, parent_positions or []))
+        target_transforms = [(x, y, z, 0.0, 0.0, 0.0, 1.0) for x, y, z in (parent_positions or [])]
+        cls._register_clone_transforms(source, targets, target_transforms)
```
[the remark this answers is not in the document]
Comment (marcodiiga) [reply source/isaaclab_ovphysx/isaaclab_ovphysx/physics/ovphysx_manager.py:420]: Kept the hook private because this is package-internal plumbing and exposing it would enlarge the supported API without a user need. Addressed the type concern with a fixed seven-float `_CloneTransform`; the public translation-only `register_clone` interface remains unchanged.

File: source/isaaclab_ovphysx/test/test_cloner.py [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+    context = OvPhysxReplicateContext(stage)
+    context.queue_mapping(
+        sources=["/World/envs/env_0/Robot", "/World/envs/env_9/Inactive"],
+        destinations=["/World/envs/env_{}/Robot", "/World/envs/env_{}/Inactive"],
+        env_ids=torch.tensor([0, 1]),
+        mapping=torch.tensor([[True, True], [False, False]]),
+        positions=torch.tensor([[4.0, 5.0, 6.0], [10.0, 20.0, 30.0]]),
+        quaternions=torch.tensor([[0.0, 0.0, 0.0, 1.0], [0.0, 0.0, half_sqrt_two, half_sqrt_two]]),
+    )
+
```
[the remark this answers is not in the document]
Comment (marcodiiga) [reply source/isaaclab_ovphysx/test/test_cloner.py:53]: Addressed: the nested regression now calls `replicate()`, asserts the context queue is cleared, and verifies the exact seven-value transforms reach the manager. A companion test preserves the translation-only `register_clone` compatibility behavior.

File: source/isaaclab_ovphysx/isaaclab_ovphysx/cloner/replicate.py [hunk 8]
```diff
@@ -24,10 +24,12 @@
 
 import torch
 
-from pxr import Sdf, Usd
+from pxr import Gf, Sdf, Usd, UsdGeom
 
 from isaaclab.cloner.cloner_utils import split_clone_template
 
+_CloneTransform = tuple[float, float, float, float, float, float, float]
```
Comment (AntoineRichard) [inline source/isaaclab_ovphysx/isaaclab_ovphysx/cloner/replicate.py]: **Maintainability:** Define the clone payload once. `_CloneTransform` is duplicated verbatim in `ovphysx_manager.py`, and both modules separately convert `(x, y, z)` to `(x, y, z, 0, 0, 0, 1)`. Put the alias and conversion helper in one private module and use it from both call paths so this cross-module wire format cannot drift.

File: source/isaaclab_ovphysx/isaaclab_ovphysx/cloner/replicate.py [hunk 9]
```diff
@@ -60,9 +79,18 @@ def queue(
         Args:
             source: Source prim path.
             targets: Destination prim paths.
-            parent_positions: Parent Xform positions [m] for each destination.
+            parent_positions: Legacy translation-only world positions [m] for
```
Comment (AntoineRichard) [inline source/isaaclab_ovphysx/isaaclab_ovphysx/cloner/replicate.py]: **Documentation:** `queue()` remains a supported public method; nothing deprecates it. Calling this input “Legacy” is inaccurate and reads as migration history. Describe only the current contract, for example: “Final world positions [m] for each target root; orientations are identity.”

File: source/isaaclab_ovphysx/isaaclab_ovphysx/cloner/replicate.py [hunk 10]
```diff
@@ -82,41 +110,62 @@ def queue_mapping(
...
+            if not active_env_ids:
+                continue
 
             self_env_id: int | None = None
             pre, suf = split_clone_template(destinations[i])
             candidate = src.removeprefix(pre).removesuffix(suf)
             if candidate.isdigit():
                 self_env_id = int(candidate)
 
+            source_world = xform_cache.GetLocalToWorldTransform(self.stage.GetPrimAtPath(src)).RemoveScaleShear()
```
Comment (AntoineRichard) [inline source/isaaclab_ovphysx/isaaclab_ovphysx/cloner/replicate.py]: **Correctness:** Validate the source prim and derived anchor before passing them to `XformCache`. `GetLocalToWorldTransform()` returns identity for an invalid prim, so a stale source/template silently queues wrongly placed clones. Since `OvPhysxReplicateContext` is exported, enforce this invariant at the boundary with a clear `ValueError`.

File: source/isaaclab_ovphysx/isaaclab_ovphysx/cloner/replicate.py [hunk 11]
```diff
@@ -82,41 +110,62 @@ def queue_mapping(
...
                     continue
                 targets.append(destinations[i].format(env_id))
-                if positions is not None and env_id < len(positions):
-                    pos = positions[env_id]
-                    parent_positions.append((float(pos[0]), float(pos[1]), float(pos[2])))
-                else:
-                    parent_positions.append((0.0, 0.0, 0.0))
+
+                target_env_world = Gf.Matrix4d(1.0)
+                if positions_list is not None and env_id < len(positions_list):
```
Comment (AntoineRichard) [inline source/isaaclab_ovphysx/isaaclab_ovphysx/cloner/replicate.py]: **Correctness:** `None` may mean identity, but a provided pose tensor that lacks an active `env_id` should not also mean identity. Validate both provided tensors against the selected IDs before this loop and raise on malformed input; otherwise a short tensor silently collapses a clone toward the origin or default orientation.

File: source/isaaclab_ovphysx/isaaclab_ovphysx/cloner/replicate.py [hunk 12]
```diff
@@ -145,13 +194,10 @@ def ovphysx_replicate(
...
     has been attached.
 
-    The ``positions`` parameter contains the 2-D grid world positions for all
-    environments.  They are forwarded to the C++ clone plugin so that the
-    parent Xform prim for each clone (e.g. ``/World/envs/env_N``) is placed at
-    the correct grid location in Fabric. The in-memory OVStage only contains
-    ``env_0``; without explicit positions all clone parents would be created at
-    the origin, causing all articulations to pile up and the GPU solver to
-    diverge on the first warmup step.
+    The ``positions`` and ``quaternions`` parameters describe each environment's
```
Comment (AntoineRichard) [inline source/isaaclab_ovphysx/isaaclab_ovphysx/cloner/replicate.py:224]: **Documentation:** This repeats the module docstring’s manager/clone lifecycle and restates implementation details. Keep this function docstring at the public contract: source-relative rows are composed with target environment poses and queued. Let the module docstring own the deferred replay explanation.

File: source/isaaclab_ovphysx/isaaclab_ovphysx/physics/ovphysx_manager.py [hunk 13]
```diff
@@ -214,11 +216,10 @@ class OvPhysxManager(PhysicsManager):
     # :meth:`_release_physx`); we mirror it here so a clear Python error is raised
     # if a later :class:`~isaaclab.sim.SimulationContext` requests a different device.
     _locked_device: ClassVar[str | None] = None
-    # Pending (source, targets, parent_positions) triples queued by
-    # replication is queued before the PhysX instance exists.  Replayed via
-    # physx.clone() in _warmup_and_load().
-    # parent_positions is a list of (x, y, z) tuples — one per target.
-    _pending_clones: ClassVar[list[tuple[str, list[str], list[tuple[float, float, float]]]]] = []
+    # Pending (source, targets, target_transforms) triples queued before the
```
Comment (AntoineRichard) [inline source/isaaclab_ovphysx/isaaclab_ovphysx/physics/ovphysx_manager.py]: **Comment hygiene:** The annotation and `_CloneTransform` already define the tuple contents. Replace these three lines with the durable lifecycle fact only, such as `# Clone operations queued until the PhysX instance is initialized.` This avoids duplicating a wire-format description that must stay synchronized.

File: source/isaaclab_ovphysx/isaaclab_ovphysx/physics/ovphysx_manager.py [hunk 14]
```diff
@@ -243,23 +244,28 @@ def fix_articulation_root(cls, articulation_prim: Any, stage: Any = None) -> Any
...
         cls, source: str, targets: list[str], parent_positions: list[tuple[float, float, float]] | None = None
     ) -> None:
-        """Register a (source, targets, parent_positions) triple for replay via physx.clone().
+        """Register translation-only whole-environment clones for replay via ``physx.clone()``.
 
-        Called by :func:`~isaaclab_ovphysx.cloner.ovphysx_replicate` during
-        scene setup, before the PhysX instance exists.  The clone operations
-        are executed in :meth:`_warmup_and_load` immediately after
-        attaching the populated OVStage.
+        This compatibility entry point accepts positions only. The full-pose
```
Comment (AntoineRichard) [inline source/isaaclab_ovphysx/isaaclab_ovphysx/physics/ovphysx_manager.py]: **Documentation:** Keep this public docstring functional. “Compatibility entry point,” the private-method reference, and the warmup narrative describe internal history and plumbing already covered elsewhere. State that the method queues clones at the given world positions with identity rotations, then retain the `Args` block.

File: source/isaaclab_ovphysx/test/test_cloner.py [hunk 15]
```diff
@@ -0,0 +1,88 @@
...
+    context.queue_mapping(
+        sources=["/World/envs/env_0/Robot", "/World/envs/env_9/Inactive"],
+        destinations=["/World/envs/env_{}/Robot", "/World/envs/env_{}/Inactive"],
+        env_ids=torch.tensor([0, 1]),
+        mapping=torch.tensor([[True, True], [False, False]]),
+        positions=torch.tensor([[4.0, 5.0, 6.0], [10.0, 20.0, 30.0]]),
+        quaternions=torch.tensor([[0.0, 0.0, 0.0, 1.0], [0.0, 0.0, target_half_angle_sin, target_half_angle_cos]]),
+    )
+
+    assert len(context._queue) == 1
```
Comment (AntoineRichard) [inline source/isaaclab_ovphysx/test/test_cloner.py]: **Test design:** This asserts `context._queue`, calls `replicate()`, then asserts the same payload again in `_pending_clones`. Remove the intermediate private-state block and assert the expected transform once at the handoff—via a spy on `_register_clone_transforms` or against the manager queue after `replicate()`—plus queue clearing. This preserves coverage without duplicated, implementation-coupled assertions.

**Review by AntoineRichard — CHANGES_REQUESTED** [review]
The pose composition is coherent, the quaternion convention matches sibling cloners, and the regression is effective. I am requesting changes for the remaining silent-invalid-input paths and showroom-quality cleanup called out inline. The branch also currently conflicts with `develop` and must be updated before merge.

Verification: the two focused tests pass on this head and both fail against the base implementation; the full pre-commit suite passes.

**Review by AntoineRichard — APPROVED** [review]
